src/os/JournalingObjectStore.cc

```diff
@@ -193,27 +198,38 @@ bool JournalingObjectStore::ApplyManager::commit_start()
     dout(10) << "commit_start max_applied_seq " << max_applied_seq
 	     << ", open_ops " << open_ops
 	     << dendl;
+  
     blocked = true;
-    while (open_ops > 0) {
-      dout(10) << "commit_start waiting for " << open_ops << " open ops to drain" << dendl;
-      blocked_cond.Wait(apply_lock);
-    }
-    assert(open_ops == 0);
-    dout(10) << "commit_start blocked, all open_ops have completed" << dendl;
     {
       Mutex::Locker l(com_lock);
-      if (max_applied_seq == committed_seq) {
-	dout(10) << "commit_start nothing to do" << dendl;
+
+      set<uint64_t>::iterator it = applying_seq_set.begin();
+      if(it == applying_seq_set.end()){
+       		_committing_seq = committing_seq = max_applied_seq; 
+		
+        	dout(10) << "commit_start committing from end: " << committing_seq 
+			 << ",current applying_seq_set size:"<< applying_seq_set.size()
+			<< ", still blocked" << dendl;
+		assert(applying_seq_set.size() ==0);
+      }else
+      {
+        	_committing_seq = committing_seq = *it;
```

Comment:
yes, you are right ,it should be 
    _committing_seq = committing_seq = *it-1;